File: frontend/the_deck/lib/Models/Product.dart
class Product {
  final int id;
  final String name;
  final String image;
  final String description;
  final double price;
  final int CategoryID;

  Product({
    required this.id,
    required this.name,
    required this.image,
    required this.price,
    required this.description,
    required this.CategoryID,
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['id'],
      name: json['name'],
      image: json['image'],
      price: double.parse(json['price']),
      description: json['description'],
      CategoryID: json['category_id'],
    );
  }
}

File: The Deck/Models/entity/Product.go
package entity

import "time"

type Product struct {
	Id          uint      `json:"id"`
	Name        string    `json:"name" gorm:"type:varchar(50);index:idx_nm,unique" validate:"required"`
	Description string    `json:"description" gorm:"type:text" validate:"required"`
	Image       string    `json:"image" gorm:"type:varchar(50)" validate:"required"`
	Price       string    `json:"price" gorm:"type:varchar(50)" validate:"required"`
	CategoryID  uint      `json:"category_id" gorm:"index"`
	Category    Category  `json:"-" gorm:"foreignKey:category_id"`
	CreatedAt   time.Time `json:"created_at" gorm:"autoCreateTime" db:"created_at"`
	UpdatedAt   time.Time `json:"updated_at" gorm:"autoUpdateTime" db:"updated_at"`
	AdminID     uint      `json:"admin_id" gorm:"index"`
	Admin       Admin     `json:"-" gorm:"foreignKey:admin_id"`
}
